core/src/display_object/loader_display.rs

```diff
@@ -180,10 +180,24 @@ impl<'gc> TInteractiveObject<'gc> for LoaderDisplay<'gc> {
         point: Point<Twips>,
         require_button_mode: bool,
     ) -> Option<InteractiveObject<'gc>> {
+        // Don't do anything if run in an AVM2 context.
+        if self.as_displayobject().movie().is_action_script_3() {
+            return None;
+        }
+
         for child in self.iter_render_list().rev() {
             if let Some(int) = child.as_interactive() {
                 if let Some(result) = int.mouse_pick_avm1(context, point, require_button_mode) {
-                    return Some(result);
+                    if int.as_displayobject().movie().is_action_script_3() {
+                        let avm2_result = int.mouse_pick_avm2(context, point, require_button_mode);
+                        if let Avm2MousePick::Hit(result) = avm2_result {
+                            return Some(result);
+                        } else {
+                            return None;
```

Comment:
The old code in this block did not appear to ever return `None`. Is it correct to return `None` in this case?